x509/x509.go

```diff
@@ -557,6 +566,18 @@ func namedCurveFromOID(oid asn1.ObjectIdentifier) elliptic.Curve {
 		return elliptic.P384()
 	case oid.Equal(OIDNamedCurveP521):
 		return elliptic.P521()
+	case oid.Equal(OIDNamedCurveP192):
+		initonce.Do(func() {
```

Comment:
Since this but of functionality closely matches the `elliptic` package format, consider factoring out the curve definition into a separate package for testing, reusability, and potential inclusion into golang/x/crypto.

Reply:
I've pulled the curve stuff out into separate files but I'd prefer to avoid the overhead of an entire new package (plus the dependency management thereof) for ~30 lines of code.